service/matching/task_validation.go

```diff
@@ -121,6 +145,29 @@ func (v *taskValidatorImpl) preValidate(
 	return time.Since(v.lastValidatedTaskInfo.validationTime) > taskReaderValidationThreshold
 }
 
+// preValidatePassive track a task and return if validation should be done, if namespace is passive
+func (v *taskValidatorImpl) preValidatePassive(
+	task *persistencespb.AllocatedTaskInfo,
+) bool {
+	if v.lastValidatedTaskInfo.taskID != task.TaskId {
```

Comment:
might be nice to factor out the duplicate code? unless you're planning more changes to this logic.

the difference is just whether we skip validation the first time we see a task, right?

Reply:
>the difference is just whether we skip validation the first time we see a task, right?

yes

>might be nice to factor out the duplicate code? unless you're planning more changes to this logic.

usually yes, but this case i would argue no
1. only few lines of duplicated code
2. active & passive logic are different, this impl will be easier to understand compared to inserting `if else`